dev/devicelab/bin/tasks/hello_world_impeller.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async' show StreamSubscription;
+import 'dart:io' show Directory, Process;
+
+import 'package:flutter_devicelab/framework/devices.dart'
+    show Device, DeviceOperatingSystem, deviceOperatingSystem, devices;
+import 'package:flutter_devicelab/framework/framework.dart' show task;
+import 'package:flutter_devicelab/framework/task_result.dart' show TaskResult;
+import 'package:flutter_devicelab/framework/utils.dart'
+    show dir, flutter, flutterDirectory, inDirectory, startFlutter;
+import 'package:path/path.dart' as path;
+
+Future<TaskResult> run() async {
+  deviceOperatingSystem = DeviceOperatingSystem.android;
+  final Device device = await devices.workingDevice;
+  await device.unlock();
+  final Directory appDir =
+      dir(path.join(flutterDirectory.path, 'examples/hello_world'));
+
+  bool isUsingValidationLayers = false;
+  bool hasValidationErrors = false;
+  int impellerBackendCount = 0;
+
+  await inDirectory(appDir, () async {
+    await flutter('packages', options: <String>['get']);
+
+    final StreamSubscription<String> adb = device.logcat.listen(
+      (String data) {
+        if (data.contains('Using the Impeller rendering backend')) {
+          // Sometimes more than one of these will be printed out if there is a
+          // fallback.
+          impellerBackendCount += 1;
+        }
+        if (data.contains(
+            'Using the Impeller rendering backend (Vulkan with Validation Layers)')) {
+          isUsingValidationLayers = true;
+        }
+        // "ImpellerValidationBreak" comes from the engine:
+        // https://github.com/flutter/engine/blob/4160ebacdae2081d6f3160432f5f0dd87dbebec1/impeller/base/validation.cc#L40
+        if (data.contains('ImpellerValidationBreak')) {
+          hasValidationErrors = true;
+        }
+        print('something: $data');
+      },
+    );
+
+    final Process process = await startFlutter(
+      'run',
+      options: <String>[
+        '--enable-impeller',
+        '-d',
+        device.deviceId,
+      ],
+    );
+
+    // Since we are waiting for the lack of errors, there is no determinate
+    // amount of time we can wait.
+    await Future<void>.delayed(const Duration(seconds: 30));
```

Comment:
Had another thought. Is it possible to watch the logs for the app to print a message indicating that it's time to stop the test?

Reply:
Yea, but then you have the same problem, just in the flutter app not the test script.  How long do you wait to send the message?  Or what signal do you use to send it?  Something like "pump and settle" might work. I'm not sure if that's available to us though.  Maybe with a driver test?

If the problem we are trying to avoid is a test that is consistently reporting a false positives, it seems unlikely with a 30s timeout and a check for the backend message.  Maybe it could miss once?  I could make it wait for the backend error message indefinitely, then wait for errors for a certain amount of time after that.  That would reduce the chance of a false positive.

Reply:
From the perspective of the style guide, the app under test sending a message after N seconds is preferable to the test harness timing out after N seconds. In the former case, that's just how the app behaves and the test harness reacts appropriately. In the latter case, if the app behaves differently sometimes, but sometimes not, then the test would be flaky.

Reply:
Okay, it is waiting indefinitely for the backend log now, PTAL.  So, when we start waiting for any errors to pop up the engine is guaranteed to be engaged.  There can be no false negatives from some increased latency in the setup of the test.